src/extractcode/extract.py

```diff
@@ -178,16 +178,22 @@ def extract_file(location, target, kinds=extractcode.default_kinds, verbose=Fals
                      + getattr(extractor, '__module__', '')
                      + '.' + getattr(extractor, '__name__', ''))
     if extractor:
-        yield ExtractEvent(location, target, done=False, warnings=[], errors=[])
+        yield ExtractEvent(location, target if not replace_originals else location, done=False, warnings=[], errors=[])
```

Comment:
Why would the `target` (e.g. where we extract) become the `location` (what we extract) with `replace_originals`? that's mostly informative because that what events are about... but still?

Reply:
As far as I understand the code, the variables could be the following:
- `location=a/a.tar.gz`
- `target=a/a.tar.gz-extract`

Since my changes extract the content of the file `location` to the folder `location` instead of the folder `target`. I thus think that this change here is correct.